### PR TITLE
feat(tools): add flag for running single test on rwd-22

### LINKED ISSUE #47057: it's not possible to run the tests for a single challenge in RWD-22
In 2022/responsive-web-design there is no way to test for a single challenge.

As there is a step 1 for each practice project, running 
```bash
cd curriculum
npm run test -- -g 'Step 1'
```

as instructed runs a lot of tests

```bash
> @freecodecamp/curriculum@0.0.0-next.4 test
> ts-node ../node_modules/mocha/bin/mocha --delay --exit --reporter progress --bail "-g" "Step 1"

⠦ Populate tests.


  2814 passing (34s)
  3 pending
```

### PR DESCRIPTION
**Pull request by tom-raley**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

Closes #47057

I think this covers what the issue was about, but it's possible I either did this wrong or missed something. I tried to mimic the current filters best I could and running a single test now works. Open to any suggestions though!

### CODE REVIEW

File: curriculum/test/test-challenges.js [hunk 1]
```diff
@@ -132,7 +132,14 @@ setup()
...
-  if (process.env.npm_config_superblock && process.env.npm_config_block) {
+  if (
+    (process.env.npm_config_superblock && process.env.npm_config_block) ||
+    (process.env.npm_config_superblock && process.env.npm_config_filename) ||
+    (process.env.npm_config_block && process.env.npm_config_filename) ||
+    (process.env.npm_config_block &&
+      process.env.npm_config_filename &&
+      process.env.npm_config_superblock)
+  ) {
     throw new Error(`Please do not use both a block and superblock as input.`);
```
Comment (naomi-lgbt) [inline curriculum/test/test-challenges.js:143]:
```suggestion
    throw new Error(`Please do not use more than one of: superblock, block, filename`);
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Quick fix for the error message.

File: curriculum/test/test-challenges.js [hunk 2]
```diff
@@ -132,8 +132,17 @@ setup()
...
 
 async function setup() {
-  if (process.env.npm_config_superblock && process.env.npm_config_block) {
-    throw new Error(`Please do not use both a block and superblock as input.`);
+  if (
+    (process.env.npm_config_superblock && process.env.npm_config_block) ||
+    (process.env.npm_config_superblock && process.env.npm_config_filename) ||
+    (process.env.npm_config_block && process.env.npm_config_filename) ||
+    (process.env.npm_config_block &&
+      process.env.npm_config_filename &&
```
Comment (moT01) [inline curriculum/test/test-challenges.js:140]: Here, you are looking at `filename`, but below you are checking for `dashedname`. I presume you initially wanted to use the filename, but then noticed that info wasn't part of the challenge info.

I would use the challenge `id`. Many of the new RWD challenges have the same name - e.g. `step-1`, `step-2` - so `dashedname` wouldn't really work for those. The `id` should work for any type of challenge.

**Review by moT01 — CHANGES_REQUESTED** [review]

File: curriculum/test/test-challenges.js (hunk 2 above)
Comment (tom-raley) [reply curriculum/test/test-challenges.js]: That is indeed what happened 😄  Sorry, was out of town and didn't have a computer, but the PR has been updated and should address this, thanks for the feedback!

**Comment by naomi-lgbt** [comment]
image

Looks like it doesn't populate correctly for the step based challenges? 🤔

**Comment by naomi-lgbt** [comment]
Heya @tom-raley were you still interested in working on this?

**Comment by Sembauke** [comment]
Hey @tom-raley,

I will be closing this due inactivity.

If you need help let us know.

Have a nice day, and happy coding!

**Pull request closed without merging**